### PR TITLE
138 display pos refactor

### PR DESCRIPTION
**Pull request by alishaevn**
# Story

refactor of #254 

# Screenshots / Video
_I'm not sure why that quick error notice shows up when going back to view the browse page. will double check if it's still there after we switch to the beachside vendor with the appropriate permissions_

https://user-images.githubusercontent.com/29032869/223574402-142ee474-61ee-419f-8166-4d660f6216d0.mp4

### CODE REVIEW

File: pages/requests/[uuid].js [hunk 1]
```diff
@@ -40,27 +40,24 @@ const Request = () => {
...
-    const getPOsAsync = async () => {
-      const response = await getAllPOs(request?.quotedWareID, uuid, request?.identifier, session?.accessToken)
-      if (response.allPOs) {
-        setAllPOs(response.allPOs)
-      } else if (response.isPOError) {
-        setIsPOError(response.isPOError)
-      } else {
-        setIsPOLoading(true)
-      }
+    if (request) {
```
Comment (alishaevn) [inline pages/requests/[uuid].js:47]: making sure the request exists first so the id and identifier aren't undefined.

File: pages/requests/[uuid].js [hunk 2]
```diff
@@ -40,27 +40,24 @@ const Request = () => {
   const { messages, isLoadingMessages, isMessagesError, mutateMessages, messagesData } = useMessages(uuid, session?.accessToken)
   const { files, isLoadingFiles, isFilesError, mutateFiles, filesData } = useFiles(uuid, session?.accessToken)
 
-  const [allPOs, setAllPOs] = useState(false)
+  const [allPOs, setAllPOs] = useState([])
```
Comment (alishaevn) [inline pages/requests/[uuid].js:43]: setting this by default to an empty array (which is the value the purchase orders endpoint returns where there are no po's), allows us to define `documents` in one line.

File: pages/requests/[uuid].js [hunk 3]
```diff
@@ -40,27 +40,24 @@ const Request = () => {
...
   const [isPOError, setIsPOError] = useState(false)
-  const [isPOLoading, setIsPOLoading] = useState(false)
+  const [isLoadingPOs, setIsLoadingPOs] = useState(true)
```
Comment (alishaevn) [inline pages/requests/[uuid].js:45]: name uniformity

File: pages/requests/[uuid].js [hunk 4]
```diff
@@ -40,27 +40,24 @@ const Request = () => {
...
-      const response = await getAllPOs(request?.quotedWareID, uuid, request?.identifier, session?.accessToken)
-      if (response.allPOs) {
-        setAllPOs(response.allPOs)
-      } else if (response.isPOError) {
-        setIsPOError(response.isPOError)
-      } else {
-        setIsPOLoading(true)
-      }
+    if (request) {
+      (async () => {
```
Comment (alishaevn) [inline pages/requests/[uuid].js:48]: using an IIFE so we don't have to declare and then call a function.

File: pages/requests/[uuid].js [hunk 5]
```diff
@@ -40,27 +40,24 @@ const Request = () => {
...
-  }, [allPOs, isPOError, request?.quotedWareID, uuid, request?.identifier, session?.accessToken])
-  
-  const isLoading = isLoadingRequest || isLoadingSOWs || isLoadingFiles || isLoadingMessages || isPOLoading
-  const isError = isRequestError || isSOWError || isFilesError|| isMessagesError
-  let documents = (allSOWs) ? [...allSOWs] : []
-  allPOs ? (documents = [...documents, ...allPOs]) : (documents = [...documents])
+  }, [allPOs, isPOError, request, uuid, session])
+
+  const isLoading = isLoadingRequest || isLoadingSOWs || isLoadingFiles || isLoadingMessages || isLoadingPOs
+  const isError = isRequestError || isSOWError || isFilesError|| isMessagesError || isPOError
```
Comment (alishaevn) [inline pages/requests/[uuid].js:59]: using "isPOError"

File: utils/api/configurations.js [hunk 6]
```diff
@@ -185,37 +185,45 @@ export const  configureFiles = (data) => {
   return allFiles
 }
 
-export const configureDocuments = (documents, requestIdentifier) => {
-  return documents?.map(document => ({
-    identifier: document.identifier,
+export const configureDocument = (document, requestIdentifier) => {
```
Comment (alishaevn) [inline utils/api/configurations.js:188]: this returns one document at a time now.

File: utils/api/configurations.js [hunk 7]
```diff
@@ -185,37 +185,45 @@ export const  configureFiles = (data) => {
...
-      text: document.ship_from?.text,
+      organizationName: document.ship_from.organization_name,
+      text: document.ship_from.text,
     },
-    // the following properties only need to exist on POs
-    turnaroundTime: document.turn_around_time.human || null,
-    poNumber: document.po_number || null,
-    relatedSOWIdentifier: document.proposal_ref?.identifier || null,
-    adPO: document.scientist_identifier || null,
+    subtotalPrice: document.retail_subtotal_price_currency,
```
Comment (alishaevn) [inline utils/api/configurations.js:207]: here and below is alphabetizing the properties.

File: utils/api/configurations.js [hunk 8]
```diff
@@ -185,37 +185,45 @@ export const  configureFiles = (data) => {
...
+    taxAmount: document.tax_cost_currency,
+    terms: document.payment_terms,
+    totalPrice: document.retail_total_price_currency,
+  }
+}
+
+export const configureSOWs = (sows, requestIdentifier) => {
+  return sows?.map((sow) => ({
+    ...configureDocument(sow, requestIdentifier),
```
Comment (alishaevn) [inline utils/api/configurations.js:216]: although the sow is the same as the document at this time, this distinction will make it easier for us going forward as we add more document types.

and in the event the sow properties change, we can change them here.

**Review by summer-cook — APPROVED** [review]

**Pull request merged**